### ISSUE TITLE
Schema files from the classpath

### ISSUE THREAD

**Issue by markwimpory** [JetBrains/js-graphql-intellij-plugin#669]
Runtime and code gen time allow schema files to be loaded from the classpath including jars.  Can the plugin do this as well so that we can still get completion etc.

**Comment by FyiurAmron** [comment]
@markwimpory TBH, it's possible to work around this (I'm using this ATM in my repos), but it's tricky:
 * create a Gradle task that will extract the schemas from the artifacts you need (I use custom `Configuration` for that), into some reasonable location (build dir subdir e.g.)
 * use `graphql.config.yaml` or equivalent to provide path to that location with additional schema(s) to the plugin
 
 First-class support would be obviously a lot better here, since adding custom build jobs just to satisfy plugin requirements is somewhat unappealing. (in my case, it's to help front-end devs as well, so I'm kinda OK with it, but I wouldn't do that otherwise probably)

**Comment by vepanimas** [comment]
@markwimpory @FyiurAmron hi! I'm working on that now. Could you provide examples of how these libraries are structured in your project? If they are available in Maven, it would be even better if you could provide their IDs for testing purposes.

**Comment by FyiurAmron** [comment]
@vepanimas the structure needed to replicate this is the most bare-bones one ever:
1. create a project with an example `graphqls` in its resources. Nothing more, just build a JAR out of it, e.g. (courtesy of ):
```
interface Character {
  id: ID!
}
```
2. create a project which consumes the above JAR, e.g. via Gradle `implementation(files())` or any other mechanism, and has a second, extending `graphqls`, e.g. 
```
type Human implements Character {
  id: ID!
  name: String!
}
```
; observe that it breaks the plugin's parsing due to not scanning the classpath for the dependency `graphqls` providing the base definitions.

Basically, those two projects with those two files by themselves would be enough to replicate the core of the issue IMVHO.

**Comment by vepanimas** [comment]
[Fixed](https://youtrack.jetbrains.com/issue/WEB-64193/GraphQL-locate-schemas-in-java-libraries), should be available in the next 2024.1 EAP. Please feel free to reach out if you have any questions or feedback.
